### PR TITLE
refactor: use cloudflare provider for DNS and worker deployment

### LINKED ISSUE #98: Deploy cf worker and DNS via terraform

### PR DESCRIPTION
**Pull request by lucacasonato**
This switches to using the Cloudflare terraform provider.

These things will now be auto provisioned:
- api.deno.land DNS record
- cdn.deno.land DNS record
- cdn.deno.land cloudflare worker

This also removes the hardcoded s3 url from the worker.

Closes #98

### CODE REVIEW

File: worker/wrangler.toml [hunk 1]
```diff
@@ -1,16 +0,0 @@
...
-
-# default dev environment
-name = "cdn-dev"
-workers_dev = true
-
-# environment configuration
-[env.production]
-name = "cdn"
-route = "cdn.deno.land/*"
-workers_dev = false
```
Comment (ry) [inline worker/wrangler.toml:16]: So terraform takes care of updating the worker?

Comment (lucacasonato) [reply worker/wrangler.toml:16]: yes

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -11,6 +11,8 @@ jobs:
     runs-on: ubuntu-latest
     env:
       AWS_DEFAULT_REGION: us-east-1
+      CLOUDFLARE_ACCOUNT_ID: 895762025d37fc687ecd72d7cc80204a
```
Comment (wperron) [inline .github/workflows/ci.yml:14]: Your account ID should probably go in GH secrets

Comment (lucacasonato) [reply .github/workflows/ci.yml:14]: Why? It was always visible in `wrangler.toml` too. It isn't sensitive as you can not do anything without the `CLOUDFLARE_API_TOKEN`.

Comment (wperron) [reply .github/workflows/ci.yml:14]: Mhm, never noticed 🤔 I just like to hide those kind of details

**Pull request merged**